pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -106,6 +111,17 @@ def configure_progress_bar(self, refresh_rate=None, process_position=0):
 
         return progress_bar_callback
 
+    def _configure_timer_callback(self, max_time: Optional[Union[str, timedelta]] = None,) -> None:
+        if max_time is None:
+            return
+        if any(isinstance(cb, Timer) for cb in self.trainer.callbacks):
+            rank_zero_info(
+                "Ignoring `Trainer(max_time=...)`, callbacks list already contains a Timer."
+            )
+            return
+        timer = Timer(duration=max_time, interval="step")
+        self.trainer.callbacks.append(timer)
```

Comment:
This pattern of specifying an argument on the trainer to generate a callback is confusing for me. This could follow the same pattern as what happened with the checkpoint callback. If this feature is really useful, we will add more options for this, and those will be encapsulated in the callback, not the trainer. Then it'll become easier for users to construct a callback and pass it to the trainer via the `callbacks` arg than via the `max_time` argument. 

With this setup, the trainer become responsible for more checking logic. Like what if the user specifies both a `max_time` and a separate Timer callback?

Reply:
Hey @ananthsub. How else would you do this ?
It is the role of the Trainer to create default callbacks. When the user want to customise the behaviour, he will provide the callback directly.

Reply:
the confusion emerges if we ever have a default value set for these options. for example for checkpointing, if I enable it, I usually need to pass a callback instance to the `callbacks` argument to the trainer. to disable it, I need to specify `checkpoint_callback=False` on the Trainer. needing to use 2 different arguments to toggle the same feature on/off is extra overhead for users

I think this callback could be added first without adding the trainer argument. If there's significant demand for the trainer to configure the callback via a Traner constructor argument, we can add it then. What do you think of staging it like this?

Reply:
that's how I had it first :) the trainer argument was requested for convenience and to be grouped with the rest of stopping arguments (max_steps etc.). I will request comment from team on Monday.